wicket-core/src/test/java/org/apache/wicket/csp/CSPSettingRequestCycleListenerTest.java

```diff
@@ -0,0 +1,477 @@
+			new ContentSecurityPolicyEnforcer(tester.getApplication());
+		Assertions.assertThrows(IllegalArgumentException.class, () -> {
+			cspListener.blocking().add(REPORT_URI, "");
+		});
+	}
+
+	@Test
+	public void testInvalidReportUriInputIsRejected()
+	{
+		ContentSecurityPolicyEnforcer cspListener =
+			new ContentSecurityPolicyEnforcer(tester.getApplication());
+		Assertions.assertThrows(IllegalArgumentException.class, () -> {
+			cspListener.blocking().add(REPORT_URI, "abc?^()-_\'xyz");
+		});
+	}
+
+	@Test
+	public void testAllCSPSrcDefaultEnumsAreSetCorrectly() throws NoSuchAlgorithmException
+	{
+		ContentSecurityPolicyEnforcer cspListener =
+			new ContentSecurityPolicyEnforcer(tester.getApplication());
+
+		final int cspDirectiveCount = CSPDirective.values().length;
+		final int cspDirectiveSrcValueCount = CSPDirectiveSrcValue.values().length;
+		for (int i = 0; i < Math.max(cspDirectiveCount, cspDirectiveSrcValueCount); i++)
+		{
+			final CSPDirective cspDirective = CSPDirective.values()[i % cspDirectiveCount];
+			// FRAME-SRC wordt al gezet door de aanroep voor CHILD-SRC
+			if (!FRAME_SRC.equals(cspDirective) && cspDirective.getValue().endsWith("-src"))
+			{
+				final CSPDirectiveSrcValue cspDirectiveValue =
+					CSPDirectiveSrcValue.values()[i % cspDirectiveSrcValueCount];
+				cspListener.blocking().add(cspDirective, cspDirectiveValue);
+
+				cspListener.reporting().add(cspDirective, cspDirectiveValue);
+			}
+		}
+
+		StringBuffer headerErrors = checkHeaders(cspListener);
+
+		if (headerErrors.length() > 0)
+		{
+			Assertions.fail(headerErrors.toString());
+		}
+	}
+
+	@Test
+	public void testCSPReportUriDirectiveSetCorrectly()
+	{
+		ContentSecurityPolicyEnforcer cspListener =
+			new ContentSecurityPolicyEnforcer(tester.getApplication());
+		cspListener.blocking().add(REPORT_URI, "http://report.example.com");
+		cspListener.reporting().add(REPORT_URI, "/example-report-uri");
+
+		StringBuffer headerErrors = checkHeaders(cspListener);
+
+		if (headerErrors.length() > 0)
+		{
+			Assertions.fail(headerErrors.toString());
+		}
+	}
+
+	@Test
+	public void testCSPSandboxDirectiveSetCorrectly()
+	{
+		ContentSecurityPolicyEnforcer cspListener =
+			new ContentSecurityPolicyEnforcer(tester.getApplication());
+		final int cspSandboxDirectiveValueCount = CSPDirectiveSandboxValue.values().length;
+		for (int i = 0; i < cspSandboxDirectiveValueCount; i++)
+		{
+			final CSPDirectiveSandboxValue cspDirectiveValue = CSPDirectiveSandboxValue.values()[i];
+			if (cspDirectiveValue.equals(CSPDirectiveSandboxValue.EMPTY))
+				continue;
+
+			cspListener.blocking().add(SANDBOX, cspDirectiveValue);
+			cspListener.reporting().add(SANDBOX, cspDirectiveValue);
+		}
+
+		StringBuffer headerErrors = checkHeaders(cspListener);
+
+		if (headerErrors.length() > 0)
+		{
+			Assertions.fail(headerErrors.toString());
+		}
+	}
+
+	// FF 36+, IE (incl. Edge), Safari en Opera Mini hebben nog geen (volledige)

```

Comment:
OMG Dutch! :-)

Reply:
That's what you get when you copy large bodies of code from a Dutch codebase :) Some Dutch comments slipped through.